src/Application/MicroPresenter.php

```diff
@@ -93,11 +94,13 @@ public function run(Application\Request $request)
 			$response = array($response, array());
 		}
 		if (is_array($response)) {
-			$response = $this->createTemplate()->setParameters($response[1]);
-			if ($response[0] instanceof \SplFileInfo) {
-				$response->setFile($response[0]);
+			list($_content, $_parameters) = $response;
+			$response = $this->createTemplate()->setParameters($_parameters);
+			if ($_content instanceof \SplFileInfo) {
+				$response->setFile($_content);
 			} else {
-				$response->setSource($response[0]);
+				$response->getLatte()->setLoader(new Latte\Loaders\StringLoader);
+				$response->setFile($_content);
```

Comment:
This could be simplified:

~~~php
if (is_string($_content)) {
    $response->getLatte()->setLoader(new Latte\Loaders\StringLoader);
}
$response->setFile($_content);
~~~